app/services/invoices/create_pay_in_advance_charge_service.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+module Invoices
+  class CreatePayInAdvanceChargeService < BaseService
+    def initialize(charge:, event:, timestamp:)
+      @charge = charge
+      @event = event
+      @timestamp = timestamp
+
+      super(nil)
+    end
+
+    def call
+      ActiveRecord::Base.transaction do
+        @invoice = Invoice.create!(
+          organization: event.organization,
+          customer:,
+          issuing_date:,
+          invoice_type: :subscription,
+          payment_status: :pending,
+          currency: customer.currency,
+          taxes_rate: customer.applicable_vat_rate,
+          timezone: customer.applicable_timezone,
+          status: :finalized,
+        )
+
+        InvoiceSubscription.create!(
+          invoice:,
+          subscription: event.subscription,
+          recurring: false,
+        )
+
+        create_fees(invoice)
+
+        invoice.fees_amount_cents = invoice.fees.sum(:amount_cents)
+        invoice.sub_total_excluding_taxes_amount_cents = invoice.fees.sum(:amount_cents)
```

Comment:
Maybe you could reuse `invoice.fees_amount_cents` to reduce the number of query sent to the DB?

Reply:
OFC, nice one 👍